lib/date_time/date_time.c

```diff
@@ -631,19 +91,18 @@ int date_time_uptime_to_unix_time_ms(int64_t *uptime)
 
 	uptime_prev = *uptime;
 
-	if (!initial_valid_time) {
-		LOG_WRN("Valid time not currently available");
-		return -ENODATA;
+	err = date_time_now(&date_time_ms);
+	if (err) {
+		return err;
 	}
 
-	*uptime += time_aux.date_time_utc - time_aux.last_date_time_update;
+	*uptime += date_time_ms - date_time_core_last_update_uptime();
```

Comment:
This logic is incorrect. It should be subtracting the current uptime (k_uptime_get()) from the current UTC time (date_time_ms) in order to find the difference between UTC and uptime. See PR#7039 for a fix.

FWIW I think the old way of doing this was better since it's more efficient and stable.